### PR TITLE
#1183 added erf and erfc

### LINKED ISSUE #1183: Add erf and erfc functions
**Summary**
Add the error function (erf) and complementary error function (erfc) to PyBaMM.

 **Motivation**
These functions come up in some applications (e.g. Stefan problems) and it would good to have them into PyBaMM. Should be an easy implementation, however it would require calling SciPy from `functions.py`. Would that be an issue?

### PR DESCRIPTION
**Pull request by brosaplanella**
# Description

Added Erf and Erfc functions

Fixes #1183 

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/master/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [x] New feature (non-breaking change which adds functionality)

# Key checklist:

- [x] No style issues: `$ flake8`
- [ ] All tests pass: `$ python run-tests.py --unit`
- [x] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks:

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: pybamm/expression_tree/functions.py [hunk 1]
```diff
@@ -450,3 +451,35 @@ def _function_diff(self, children, idx):
...
+
+class Erfc(SpecificFunction):
+    """ Complementary error function """
+
+    def __init__(self, child):
+        super().__init__(1 - special.erf, child)
+
+    def _function_diff(self, children, idx):
+        """ See :meth:`pybamm.Function._function_diff()`. """
+        return -2 / np.sqrt(np.pi) * Exponential(-children[0] ** 2)
```
Comment (tinosulzer) [inline pybamm/expression_tree/functions.py:480]: I don't think we need this class, given how you defined `erfc`

**Review by tinosulzer — APPROVED** [review]
Looks good, just needs CHANGELOG :) and you can remove `Erfc`

**Comment by ferranbrosa** [comment]
Not sure why does the test not pass in MacOs. Maybe the values are slightly different and that's why it crashes. Can someone with a Mac check if that's the case?

**Comment by tinosulzer** [comment]
Yeah it seems like casadi and numpy have different implementations of erf on Mac:
``` 
In [5]: casadi.DM(erf(3)) - casadi.erf(3)
Out[5]: DM(-1.11022e-16)
```

**Comment by tinosulzer** [comment]
Opened https://github.com/casadi/casadi/issues/2672

**Comment by ferranbrosa** [comment]
Should I define an `assertCasadiAlmostEqual` to test it then?

**Comment by tinosulzer** [comment]
If you can figure out how :) I briefly tried but couldn't. It's still weird that casadi gives different values on mac and windows/linux though

**Comment by tinosulzer** [comment]
CasADi people say it's not something they can fix. So we should figure out how to implement assertCasadiAlmostEqual

File: tests/unit/test_expression_tree/test_operations/test_convert_to_casadi.py [hunk 2]
```diff
@@ -15,6 +16,15 @@ def assert_casadi_equal(self, a, b, evalf=False):
...
             self.assertTrue((a - b).is_zero())
 
+    def assert_casadi_almost_equal(self, a, b, decimal=7, evalf=False):
+        tol = 1.5 * 10**(-decimal)
+        if evalf is True:
+            self.assertTrue(
+                (casadi.fabs(casadi.evalf(a) - casadi.evalf(b)) < tol).is_one()
+            )
+        else:
+            self.assertTrue((casadi.fabs(a - b) < tol).is_one())
```
Comment (tinosulzer) [inline tests/unit/test_expression_tree/test_operations/test_convert_to_casadi.py:26]: nicely done!

**Pull request merged**